### ISSUE TITLE
Multiple content types

### ISSUE THREAD

**Issue by raae** [nuxt-community/nuxtent-module#113]
I must have misunderstood something as I cannot get two content types to work together.

I am trying to create both projects and posts with permalinks `:slug` using the configuration below. I would like the projects to use the page `projects/_slug` and the posts `posts\_slug`. However both use `posts\_slug` giving me a 404 on /project1 as `project1` does not exist in the posts collection.

Full example can be found at https://github.com/raae/nuxtent-test.

`module.exports = {
  content: [
    [
      'projects',
      {
        permalink: ':slug',
        page: 'projects/_slug',
        isPost: false,
        generate: [
          // for static build
          'get',
          'getAll'
        ]
      }
    ],
    [
      'posts',
      {
        permalink: ':slug',
        page: '/posts/_slug',
        isPost: false,
        generate: [
          // for static build
          'get',
          'getAll'
        ]
      }
    ]
  ]
}
`

**Comment by medfreeman** [comment]
Hi, you should use '/projects/:slug' as projects permalink, and '/posts/:slug' as posts permalink to achieve the behavior you want.
The docs aren't completely up-to-date, including about this topic.

Basically i changed the behavior a few weeks ago, so you could eventually get '/name-of-post' posts url along with '/projects/name-of-project' projects urls for example, thus removing the automatic prefixing of permalinks when having multiple content types.

**Comment by alidcastano** [comment]
That's what I figured was the problem -- since they both have same permalink, the one that is registered first overrides the other. @raae Nuxtent overrides page paths so even though you nested the pages you should, it's the permalink that ends up defining the route's behavior... Try putting `posts/:slug` and `projects/:slug` as the permalink, and let us know if that works!


--
Side discussion:
@medfreeman But why is one route ending up as `post/_slug` if both of them have `:slug` as the permalink?

Also, this is why I recommended having automatic prefixing of content routes as the default behavior. I think it's what most people would expect. If you define an index route then you'd expect a root level path, but why would you nest a page, e.g. `somePath/_slug` yet expect a root level path?

The question is: should it be expected that a page found under `nestedPath/_slug` with permalink `:slug` be found under `nestedPath/slug` OR just `slug` ?  

Similarly, should a page under `nestedPath/_slug` with permalink `:year/slug` be under `nestedPath/2017/slug` or  `2017/slug`? I could see how the latter would be more powerful, but why wouldn't the user just make the page `_slug` since you can't have more than more dynamic route anyways as they would override each other.

**Comment by raae** [comment]
So my use case is that I would like to organize the content in different folders and also have them accessible as separate collections through getAll(). But I want both to resolve to `<baseUrl>/:slug` permalink. 

But that might not be possible?

**Comment by alidcastano** [comment]
Yes this should be possible with the current implementation. Does putting the permalinks as `posts/:slug` and `projects/:slug` not work?

**Comment by raae** [comment]
Yes, it works. But the permalink is then <baseUrl>/posts/post1 and <baseUrl>/projects/project1. And the path values feel a little weird.

<img width="716" alt="nuxtent_starter" src="https://user-images.githubusercontent.com/478531/32284182-af21b60e-bf26-11e7-8586-720dc4d7827d.png">

My config is now: 
````
module.exports = {
  content: [
    [
      'projects',
      {
        permalink: 'projects/:slug',
        page: 'projects/_slug',
        isPost: false,
        generate: [
          // for static build
          'get',
          'getAll'
        ]
      }
    ],
    [
      'posts',
      {
        permalink: 'posts/:slug',
        page: '/posts/_slug',
        isPost: false,
        generate: [
          // for static build
          'get',
          'getAll'
        ]
      }
    ]
  ]
}
```

**Comment by dahrens** [comment]
I encountered the same issue - I solved it by omitting the nesting in the pages directory.

```javascript
  content: [
    ['writings', {
      permalink: ':year/:slug',
      page: '/_writings',
      generate: [
        'get', 'getAll'
      ]
    }],
    ['projects', {
      permalink: 'projects/:slug',
      isPost: false,
      page: '/_projects',
      generate: [
        'get', 'getAll'
      ]
    }]
  ],
```

Note that in pages I directly use `_projects` instead of using a subfolder.

**Comment by raae** [comment]
If you could point me in the right direction (ie. relevant code) I could give it a try.

**Comment by alidcastano** [comment]
Yeah sorry I'm just to busy to look into this week, but I come here to quickly answer questions :)

If you want to give it a go, the relevant code is at : https://github.com/nuxt-community/nuxtent-module/blob/master/lib/content/build.js#32

If you have questions feel free to also join our dev [slack channel](https://join.slack.com/t/nuxtent/shared_invite/enQtMjQ1MzI4MzgxNzI5LTcwOTlkYTczNTMzMGU1YjYyYjlkNzNjNDc3MjQ3YTdmMjc0ZThkZjM1ZWY4ZjhiZjc1MDIwYjJmMjAxODlhMzU) and ask them there

**Comment by joostdecock** [comment]
@raae I know this has been open for a while, but I think you might be confusing two things:

 - `path` : The path used to fetch the item from the content API
 - `permalink` : The path used by the (user in) the browser

So, in the screenshot above, you have a content entry with:

 - `path` : `/projects/projects/project1`  => Use this to fetch from content-api in the page for your dynamic root
 - `permalink` : `/projects/project1` => This is what the user navigates to

This is actually rather handy, because `/projects` is how the content-api groups all your projects. So in the page that handles the dynamic root for this content type, you simple fetch /projects+permalink from the content api and you'll get the content for the the page the user requested.

Just keep in mind that to get stuff from the content api, you fetch: `/content-api/[content-group]/[permalink of the content item]`

For me, this works as expected and is not a bug. Thoughts?

**Comment by renestalder** [comment]
I really don't get those multiple content yes working. I probably tried over 3 hours with different configurations.

Take this simple single content type configuration:

```js
module.exports = {
  api: {
    baseURL: process.env.NODE_ENV === 'production'
      ? 'http://v2.imquadrat.ch'
      : 'http://localhost:3000',
    browserBaseURL: process.env.NODE_ENV === 'production'
      ? 'http://v2.imquadrat.ch'
      : 'http://localhost:3000'
  },
  content: {
    page: '/_page',
    permalink: ':slug',
    isPost: false,
    generate: [ // for static build
      'get'
    ],
  }
}
```

I have several md files in then `content` folder and in subfolders. They get flattened and stored in the dist folder on `nuxt generate`.

Good so far.

Now I'm changing this to a content type:

```js
module.exports = {
  api: {
    baseURL: process.env.NODE_ENV === 'production'
      ? 'http://v2.imquadrat.ch'
      : 'http://localhost:3000',
    browserBaseURL: process.env.NODE_ENV === 'production'
      ? 'http://v2.imquadrat.ch'
      : 'http://localhost:3000'
  },
  content: [
    ['pages', {
      page: '/pages/_page',
      permalink: 'pages/:slug',
      isPost: false,
      generate: [ // for static build
        'get'
      ]
    },
  ]
}
```

Moving the files to the correct paths and try to generate or run, I always run in 404s or other errors.
Do you have any working example online which is using multiple content types, also with root paths and subfolders?

**Comment by joostdecock** [comment]
You use `pages/:slug` for the permalink, but the subdirectories are not part of the slug. As such, your permalink a don't match your directory structure, which might be causing the 404s.

Try with `permalink: '/docs/:section*/:slug',` as that should add the directory in there.

There's an example using this [here](https://github.com/freesewing/site/blob/develop/nuxtdown.config.js) but it's from the nuxtdown fork so you won't be able to just copy-paste it.

But the permalink stuff should be the same though.

**Comment by renestalder** [comment]
> You use pages/:slug for the permalink, but the subdirectories are not part of the slug

Not sure what you mean with that. In my understanding, my configuration should match content stored in `content/pages`, so with `pages/:slug` as permalink, this would match this structure.

This would mean.
Content: `content/pages/my-page.md`.
Template: `pages/pages/_page.vue`
Permalink: `mywebsite.com/my-page`

What do I misunderstand?

**Comment by joostdecock** [comment]
It's the subfolders that are a problem. What you describe above works fine if you have all your markdown in 1 folder, but not if you have subfolders.

So:

`/content/pages/my-page.md` and `/content/pages/subdir/my-page.md` have the same slug (my-page).

Your permalinks only include the slug, so the 2pages will end up with the same permalinks. You need to add in the 'sectiin' which is essentially the subdir. The example I linked to above linked this.

Either that, it it's me who misunderstands your issue 😉

**Comment by renestalder** [comment]
I feel like you misunderstand me. I don't have any subfolders in the `content/pages` folder.
Really, the only thing I did is converting my single content type example above into the multicontent type configuration without adding any other content type. For this I've put my content into `content/pages` instead of `content/` directly, reflected the same structure in `pages/` folder to easier relate. The result is always a 404 when trying to access `myurl.tld/pages/page-3`.

```
├── content/
│   └── pages
│         ├── page-1.md
│         ├── page-2.md
│         └── page-3.md
└── pages/
    └── pages
          └── _page.vue
```

If this is something that is fixed in nuxtdown, fine. But from the readme of nuxtdown, it will only be a temporary repository that is eventually going to be merged back to nuxtent. I would rather love to know how this works in nuxtent and find out where the problem is and directly fix it in nuxtent if a fix is needed.
I still believe I probably doing something wrong and if this is the case, based on the time I invested I also would love to improve the documentation if needed as soon as I understand it.

**Comment by renestalder** [comment]
@cesasol Thanks a lot for your input. I then gonna try the development branch. Already saw that there are fixes not available on the released version but wasn't sure if it is safe to try.

Let me know if you need contributors. I really would like to keep this project alive and If I can help instead of developing something from scratch for my Netlify CMS coupled websites I'll do that.

**Comment by renestalder** [comment]
@cesasol With your submodule solution, doesn't Nuxt complain about `import` statements? How you worked around this? In my understanding, Nuxt v1 doesn't support es modules.

**Comment by cesasol** [comment]
Another thing, where I work we decide to contribute with this project for the same reason as you, not having to develop a cms for static sites from scratch.
